flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -1133,15 +1134,11 @@ def fir_EmboxOp : fir_Op<"embox", [NoSideEffect, AttrSizedOperandSegments]> {
         if (lenPs != rt.getNumLenParams())
           return emitOpError("number of LEN params does not correspond"
                              " to the !fir.type type");
+      } else if (auto strTy = fir::unwrap_char(eleTy)) {
+        if (strTy.getLen() != fir::CharacterType::unknownLen())
+          return emitOpError("CHARACTER already has static LEN");
```

Comment:
This isn't checking the same thing as the original code.

FIR doesn't allow `fir.embox %1 typeparams %2 : (!fir.ref<!fir.array<12x!fir.char<1>>>, index) -> !fir.box<!fir.array<?x!fir.char<1>>>`.